parlai/agents/fairseq/fairseq.py

```diff
@@ -448,10 +451,31 @@ def eval_step(self, batch):
         if batch.label_vec is not None:
             # Interactive mode won't have a gold label
             self.trainer.valid_step(samples)
-        # Grade each of the candidate sequences
-        # TODO: grade everything in observations[i]['label_candidates']
 
-        if not self.args.skip_generation:
+        # Grade each of the candidate sequences
+        if batch.candidate_vecs is not None:
```

Comment:
Review question: would we rather this be self.rank_candidates?

Reply:
I think this is a fine way to do it: rank the candidates if they're there.